conf/exactmatch-val.bess

```diff
@@ -0,0 +1,90 @@
+# list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+# this list of conditions and the following disclaimer in the documentation
+# and/or other materials provided with the distribution.
+#
+# * Neither the names of the copyright holders nor the names of their
+# contributors may be used to endorse or promote products derived from this
+# software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+import scapy.all as scapy
+import socket
+
+src_mac='02:1e:67:9f:4d:ae'
+dst_mac='06:16:3e:1b:72:32'
+nh_mac1='30:31:32:33:34:35'
+nh_mac2='40:41:42:43:44:45'
+nh_mac3='50:51:52:53:54:55'
+nh_mac4='60:61:62:63:64:65'
+
+def aton(ip):
+    return socket.inet_aton(ip)
+
+def mac2hex(mac):
+    return long(mac.replace(':', ''), 16)
+
+# Craft a packet with the specified IP addresses
+def gen_packet(proto, src_ip, dst_ip):
+    eth = scapy.Ether(src=src_mac, dst=dst_mac)
+    ip = scapy.IP(src=src_ip, dst=dst_ip)
+    udp = proto(sport=10001, dport=10002)
+    payload = 'helloworld'
+    pkt = eth/ip/udp/payload
+    return bytes(pkt)
+
+packets = [gen_packet(scapy.UDP, '172.16.100.1', '10.0.0.1'),
+           gen_packet(scapy.UDP, '172.12.55.99', '12.34.56.78'),
+           gen_packet(scapy.UDP, '172.12.55.99', '10.0.0.1'),
+           gen_packet(scapy.UDP, '172.16.100.1', '12.34.56.78'),
+           gen_packet(scapy.UDP, '192.168.1.123', '12.34.56.78'),
+          ]
+
+# L4 protocol and source/destination IP addresses
+em::ExactMatch(fields=[{'offset':0, 'num_bytes':6},
+                       {'offset':23, 'num_bytes':1},
+                       {'offset':26, 'num_bytes':4},
+                       {'offset':30, 'num_bytes':4},
+                       {'attr_name':'foo', 'num_bytes':2}],
+               values=[{'offset':0, 'num_bytes':6},
+                       {'attr_name':'bar', 'num_bytes':5}])
+
+Source() -> Rewrite(templates=packets) -> SetMetadata(attrs=[{'name': 'foo', 'size': 2, 'value_int':0x3344}]) -> em
+
+em:0 -> Sink()
+em:1 -> Sink()
+em:2 -> Sink()
+em:3 -> Sink()
+em:4 -> Sink()
+em:5 -> Sink()   # used as default gate
```

Comment:
we should have something after em, maybe Split, to verify metadata/offset were set.

{0,1}hit {2}deleted rule {3}default

0-> Split based on attr -> x2 Sinks
1-> Split based on offset -> x2 Sinks